examples/siro_sandbox/app_states/app_state_rearrange.py

```diff
@@ -436,3 +437,53 @@ def sim_update(self, dt, post_sim_update_dict):
     def is_app_state_done(self):
         # terminal neverending app state
         return False
+
+
+class AppStateRearrangeTutorialTransition(AppState):
```

Comment:
Some background on this refactor: I am not a fan of the `SandboxDriver._app_states` logic we added. It is inflexible; it assumes we will always iterate linearly through a list of AppStates. In practice, an application that has multiple app states will need arbitrarily complex state-transition logic. I demonstrate that here with AppStateRearrangeTutorialTransition.

Reply:
I agree that we should frame it as a state machine.